tap_github/streams.py

```diff
@@ -0,0 +1,713 @@
+                records = response.json()
+                extraction_time = singer.utils.now()
+                # Loop through all records
+                for record in records:
+
+                    record['_sdc_repository'] = repo_path
+                    self.add_fields_at_1st_level(record, {})
+                    parent_record = copy.copy(record)
+
+                    with singer.Transformer() as transformer:
+                        if record.get(self.replication_keys):
+                            if record[self.replication_keys] >= max_bookmark_value:
+                                # Update max_bookmark_value
+                                max_bookmark_value = record[self.replication_keys]
+
+                            bookmark_dttm = record[self.replication_keys]
+
+                            # Keep only records whose bookmark is after the last_datetime
+                            if bookmark_dttm >= min_bookmark_value:
+
+                                if self.tap_stream_id in selected_stream_ids:
+                                    rec = transformer.transform(record, stream_catalog['schema'], metadata=metadata.to_map(stream_catalog['metadata']))
+
+                                    singer.write_record(self.tap_stream_id, rec, time_extracted=extraction_time)
+                                    counter.increment()
+
+                                for child in self.children:
+                                    if child in stream_to_sync:
+
+                                        parent_id = tuple(parent_record.get(key) for key in STREAMS[child]().id_keys)
+
+                                        self.get_child_records(client,
+                                                            catalog,
+                                                            child,
+                                                            parent_id,
+                                                            repo_path,
+                                                            state,
+                                                            start_date,
+                                                            parent_record.get(self.replication_keys),
+                                                            stream_to_sync,
+                                                            selected_stream_ids,
+                                                            parent_record = parent_record)
+                        else:
+                            LOGGER.warning("Skipping this record for %s stream with %s = %s as it is missing replication key %s.",
+                                        self.tap_stream_id, self.key_properties, record[self.key_properties], self.replication_keys)
+
+
+            # Write bookmark for incremental stream.
+            self.write_bookmarks(self.tap_stream_id, selected_stream_ids, max_bookmark_value, repo_path, state)
+
+        return state
+
+class IncrementalOrderedStream(Stream):
```

Comment:
@prijendev what is the difference between class "IncrementalStream" and "IncrementalOrderedStream"?

Reply:
IncrementalStream is the set of streams for which API returns records in random order whereas IncrementalOrderedStream is the set of streams for which API returns records in `desc` order only.